etc/afpd/dircache.c

```diff
@@ -550,6 +703,104 @@ void dircache_remove(const struct vol *vol _U_, struct dir *dir, int flags)
                && queue_count == dircache->hash_nodecount);
 }
 
+/*!
+ * @brief Remove all child entries of a directory from the dircache
+ *
+ * When a directory is renamed or moved, the full paths stored in the
+ * dircache become invalid for all child entries of the renamed dir.
+ * This function prunes orphaned child dircache entries of given dir.
+ * CNID entries use parent DIDs and name, and requre recursion to get
+ * the full path, therefore parent changes do no invalid the CNIDs.
```

Comment:
better language would be "do not invalidate the CNIDs", wouldn't it?

I normally don't nitpick typos in code comments, but this one made it a bit hard for me to parse the meaning of the sentence at first glance :)